frontend/src/components/Header/HeaderDialog.tsx

```diff
@@ -0,0 +1,51 @@
+import Button from '@mui/material/Button';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogContentText from '@mui/material/DialogContentText';
+import DialogTitle from '@mui/material/DialogTitle';
+
+interface HeaderDialogProps {
+    openLogoutModal: boolean,
+    setOpenLogoutModal: Function,
+    callback: Function,
+    dialogTitle: String,
+    dialogContent: String,
+}
+
+export default function HeaderDialog(props: HeaderDialogProps) {
+
+    const handleClickNo = () => {
+        props.setOpenLogoutModal(false);
+    };
+    const handleClickYes = () => {
+        props.setOpenLogoutModal(false);
+        props.callback()
+    };
+
+    return (
+        <div>
```

Comment:
Why is the dialog wrapped in a div?

Reply:
My bad, I used an example that had the another top-level div in the return statement which I figured didn't fly with React, but I wonder if the single-div-render-return is even necessary anymore?

Reply:
My bad, I used an example that had the another top-level div in the return statement which I figured didn't fly with React, but I wonder if the single-div-render-return is even necessary anymore?